### PR TITLE
[INLONG-10289][Agent] Update the SQLServer Source

### LINKED ISSUE #10289: [Improve][Agent] Update the SQLServer Source
### Description

the latest Agent plugin factored, SQLServer Source needs to be updated, you can refer to [the development guide](https://inlong.apache.org/zh-CN/docs/next/design_and_concept/how_to_write_plugin_agent) and [MongoDB Source](https://github.com/apache/inlong/pull/10007).

image

Requirements:
1. Optimize the logic and performance
2. A set of UTs

**This issue is a dedicated issue for the 2024 Tencent Rhino-bird Open Source Talent Training event and is only available for college students to claim**
【Claim the issue】Sign up for the corresponding issue in columns M~R of the [Research and Study Base](https://docs.qq.com/sheet/DTk9YV3pXSklZZ29D) "issue Camp" to be considered as **successfully claimed**.
【Complete the issue】For students who have claimed the issue, please also reply "Successfully claimed this issue" in the comment section of this issue; if there is no progress submitted within 7 days, including but not limited to comments, commits, or Pull Requests, it will be considered as voluntarily giving up the issue and the committee will release the issue to the next person in line.

### InLong Component

InLong Agent

### Are you willing to submit PR?

- [ ] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zoy0**
Fixes #10289 

### Motivation

Update the SQLServer Source for agent

### Modifications

Start a Debezium thread to collect change data from SQLServer and convert time fields to string. Then put sourceData into a queue. The agent will regularly read data from this queue and send it to DataProxy.

### Verifying this change

image

### CODE REVIEW

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/task/SQLServerTask.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        }
+        if (!profile.hasKey(TaskConstants.TASK_SQLSERVER_USER)) {
+            LOGGER.error("task profile needs username");
+            return false;
+        }
+        if (!profile.hasKey(TaskConstants.TASK_SQLSERVER_PASSWORD)) {
+            LOGGER.error("task profile needs password");
+            return false;
+        }
+        if (!profile.hasKey(TaskConstants.TASK_SQLSERVER_DBNAME)) {
```
Comment (justinwwhuang) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/task/SQLServerTask.java:83]: DB_NAME?

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/SQLServerSource.java [hunk 2]
```diff
@@ -17,66 +17,180 @@
...
+            props.setProperty("value.converter.schemas.enable", "false");
+            // ignore ddl
+            props.setProperty("include.schema.changes", "false");
+            // convert time to formatted string
+            props.setProperty("converters", "datetime");
+            props.setProperty("datetime.type", "org.apache.inlong.agent.plugin.utils.SQLServerTimeConverter");
+            props.setProperty("datetime.format.date", "yyyy-MM-dd");
+            props.setProperty("datetime.format.time", "HH:mm:ss");
+            props.setProperty("datetime.format.datetime", "yyyy-MM-dd HH:mm:ss");
+            props.setProperty("datetime.format.timestamp", "yyyy-MM-dd HH:mm:ss");
```
Comment (justinwwhuang) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/SQLServerSource.java:100]: Why do we need to make special adjustments to the time format? Should we keep it as it is

**Comment by zoy0** [comment]
> 运行“mvn spotless:apply”来修复这些违规行为。

done

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/SQLServerSource.java (hunk 2 above)
Comment (zoy0) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/SQLServerSource.java:100]: The time data from debezium is timestamp. If not make special adjustments, sort-flink can't deserialize it success

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/task/SQLServerTask.java (hunk 1 above)
Comment (zoy0) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/task/SQLServerTask.java]: Done

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/SQLServerSource.java (hunk 2 above)
Comment (justinwwhuang) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/SQLServerSource.java:100]: > timestamp

Confirmed, sortlink can parse timestamps

**Comment by dockerzhang** [comment]
@zoy0 please click `resolve` if you fixed the comments.

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/SQLServerSource.java (hunk 2 above)
Comment (zoy0) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/SQLServerSource.java:100]: DATE or DATETIME data coming from Debezium sql server connector are long type. But the Flink sort can't parse unix timestamp, only SQL and ISO-8601. So we should convert the time to "yyyy-MM-dd HH:mm:ss" and let flink sort can parse them

**Comment by aloyszhang** [comment]
please rebase the master to resolve conflicting files

**Comment by zoy0** [comment]
> please rebase the master to resolve conflicting files

done

**Review by aloyszhang — APPROVED** [review]

**Comment by dockerzhang** [comment]
@haifxu @justinwwhuang PTAL, thanks.

**Review by vernedeng — APPROVED** [review]

**Review by justinwwhuang — APPROVED** [review]
LGTM

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**